curifactory/record.py

```diff
@@ -324,16 +328,45 @@ def get_dir(
         os.makedirs(dir_path, exist_ok=True)
         return dir_path
 
-    def get_reference_name(self) -> str:
+    def get_reference_name(self, map=False) -> str:
         """This returns a name describing the record, in the format 'Record [index on manager] (paramset name)
 
         This should be the same as what's shown in the stage map in the output report.
         """
-        for i, record in enumerate(self.manager.records):
+        index = self.get_record_index(map)
+        paramset_name = self.args.name if self.args is not None else "None"
+        return f"Record {index} ({paramset_name})"
+
+    def get_record_index(self, map=False) -> int:
+        record_list = self.manager.records
+        if map:
+            record_list = self.manager.map.records

```

Comment:
Why not:

```python
if map:
    record_list = self.manager.map.records
else:
    record_list = self.manager.records
```
obviously it is the same, but doesn't (shallow) copy first to potentially overwrite `record_list`. More curiosity, doesn't matter too much.

Reply:
Yeah fair enough